### PR TITLE
WIP: kubelet: do not rerun init containers if any main containers have status

### PR DESCRIPTION
**Pull request by sjenning**
Using CI to test
xref https://bugzilla.redhat.com/show_bug.cgi?id=1770017
@joelsmith @derekwaynecarr @rphillips 

Will submit upstream if tests pass.

### CODE REVIEW

File: pkg/kubelet/kuberuntime/kuberuntime_container.go [hunk 1]
```diff
@@ -742,6 +742,19 @@ func findNextInitContainerToRun(pod *v1.Pod, podStatus *kubecontainer.PodStatus)
 		return nil, nil, true
 	}
 
+	// If any of the main containers have status, then all init containers must
+	// after executed at some point in the past.  However, they could be removed
+	// from the container runtime now, and if we proceed, it would appear as if they
+	// never ran and will re-execute improperly.
+	for i := range pod.Spec.Containers {
```
Comment (rphillips) [inline pkg/kubelet/kuberuntime/kuberuntime_container.go:749]: should be InitContainers on line 749 and line 750, right?

Comment (sjenning) [reply pkg/kubelet/kuberuntime/kuberuntime_container.go:749]: no, we are wondering if there is status for an _main_ containers

File: pkg/kubelet/kuberuntime/kuberuntime_container.go [hunk 2]
```diff
@@ -742,6 +742,19 @@ func findNextInitContainerToRun(pod *v1.Pod, podStatus *kubecontainer.PodStatus)
 		return nil, nil, true
 	}
 
+	// If any of the main containers have status, then all init containers must
+	// after executed at some point in the past.  However, they could be removed
```
Comment (joelsmith) [inline pkg/kubelet/kuberuntime/kuberuntime_container.go:746]: Maybe "must have executed" instead of "must after executed"?

Comment (sjenning) [reply pkg/kubelet/kuberuntime/kuberuntime_container.go:746]: ah yes, i'll fix it up before posting upstream

**Pull request closed without merging**